packages/core/lib/utils/utils.js

```diff
@@ -16,4 +16,19 @@ const extractFlags = inputArguments => {
   return inputFlags;
 };
 
-module.exports = { extractFlags };
+const detectConfigOrDefault = options => {
+  const Config = require("@truffle/config");
+
+  try {
+    return Config.detect(options);
+  } catch (error) {
+    // Suppress error when truffle can't find a config
+    if (error.message === "Could not find suitable configuration file.") {
+      return Config.default();
+    } else {
+      throw error;
+    }
+  }
+};
```

Comment:
I got this code from `truffle version` but in the `run` / `meta` refactoring, my change to use it there was reverted. I added it back so that this function is now used in both `dashboard` and `version`.

Reply:
Opened issue here: https://github.com/trufflesuite/truffle/issues/4714